gas-tank/balance-services.md

```diff
@@ -0,0 +1,554 @@
+  method: "GET",
+  headers: {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+}).then((response) => {
+  console.log(response);
+  // Handle & do something with the response
+});
+```
+
+```json [Response]
+{
+  "additionalProp1": [
+    {
+      "id": "string",
+      "address": "string",
+      "tokenAddress": "string",
+      "balance": 0,
+      "chain": "string"
+    }
+  ],
+  "additionalProp2": [
+    {
+      "id": "string",
+      "address": "string",
+      "tokenAddress": "string",
+      "balance": 0,
+      "chain": "string"
+    }
+  ],
+  ...
+}
+```
+
+:::
+
+### Get balances by address
+
+::: code-group
+
+```javascript [Request]
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+
+await fetch(`${GAS_TANK_ENDPOINT}/balances/${address}`, { // Address to retrieve balances // [!code highlight]
+  method: "GET",
+  headers: {
+    "Content-Type": "application/json"
+    "Authorization": `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+})
+  .then((response) => {
+    console.log(response);
+    // Handle & do something with the response
+  })
+```
+
+```json [Response]
+[
+  {
+    "id": "string",
+    "address": "string",
+    "tokenAddress": "string",
+    "balance": 0,
+    "chain": "string"
+  },
+  ...
+]
+```
+
+:::
+
+### Get total balances using JWT
+
+This endpoint returns a list of total balances for the specified JWT.
+
+::: code-group
+
+```javascript [Request]
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+
+await fetch(`${GAS_TANK_ENDPOINT}/balances/totals`, {
+  method: "GET",
+  headers: {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+}).then((response) => {
+  console.log(response);
+  // Handle & do something with the response
+});
+```
+
+```json [Response]
+[
+  {
+    "address": "string",
+    "symbol": "string",
+    "value": "string",
+    "decimals": 0
+  },
+  ...
+]
+```
+
+:::
+
+### Deposit balance
+
+This endpoint allows users to deposit balance to the Gas Tank platform.
+
+To deposit the balance, you need to sign the message with the private key of the address you want to deposit the balance for. `v`, `r`, `s` are the signature fields of the message signed by the address owner.
+
+You can sign a string of data using [ethers.js](https://docs.ethers.org/v5), below is an example of [how to sign a message using ethers.js](https://docs.ethers.org/v5/getting-started/#getting-started--signing) and deposit the balance.
+
+::: code-group
+
+```javascript [Request]
+import { ethers } from "ethers";
+
+const web3 = new Web3(window.ethereum);
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+const address = "0x1234567890123456789012345678901234567890"; // Address to deposit balance // [!code highlight]
+const message = web3.utils.sha3("Deposit balance to Gas Tank"); // Message to sign // [!code highlight]
```

Comment:
Should use this endpoint.

https://gas-tank.xdefi.services/docs/#/Messages/getDepositMessageController